balancer/endpointsharding/endpointsharding.go

```diff
@@ -210,6 +215,15 @@ func (es *endpointSharding) updateState() {
 		}
 	}
 
+	// If there are no pickers available yet, return a queuing picker.
+	if pickerCount == 0 {
+		es.cc.UpdateState(balancer.State{
+			ConnectivityState: connectivity.Connecting,
+			Picker:            base.NewErrPicker(balancer.ErrNoSubConnAvailable),
+		})
```

Comment:
If no children have produced a picker, why are we doing so here?

I think the safest behavior here is to either do what you said before (and go through the exercise to make "initial state == connecting" an API guarantee), or to only have this endpointsharding LB policy produce pickers in sync with its children: i.e. whenever any child updates its picker, this policy updates its combined picker.

Reply:
If we don't send a picker here, the unit test added in the PR indicates that the channel remains in IDLE mode.